unit/Unit.h

```diff
@@ -164,5 +168,8 @@ namespace unit
 		void addItem(kitten::K_GameObject* p_item);
 		void removeItem();
 		kitten::K_GameObject* getItem() const;
+
+		//register observer
+		void registerDestroy(ability::Status* p_status);

```

Comment:
Do these ever get de-registered?
Is it possible that something registers, gets destroyed, then the callback happens?

Reply:
No, It doesn't.
I will add de-registered.